mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1307,71 +1362,93 @@ impl TakerSwap {
         };
 
         let tx_hash = transaction.tx_hash();
+        let tx_hex = transaction.tx_hex().map(From::from);
         info!("Taker payment tx hash {:02x}", tx_hash);
         let tx_ident = TransactionIdentifier {
-            tx_hex: transaction.tx_hex().into(),
+            tx_hex: tx_hex.clone(),
             tx_hash,
         };
 
         let mut swap_events = vec![TakerSwapEvent::TakerPaymentSent(tx_ident)];
-        if self.ctx.use_watchers() {
-            let preimage_fut = self.taker_coin.create_taker_spends_maker_payment_preimage(
-                &self.r().maker_payment.as_ref().unwrap().tx_hex,
-                self.maker_payment_lock.load(Ordering::Relaxed) as u32,
-                self.r().other_maker_coin_htlc_pub.as_slice(),
-                &self.r().secret_hash.0,
-                &self.unique_swap_data()[..],
-            );
-
-            match preimage_fut.compat().await {
-                Ok(preimage) => {
-                    let watcher_data = self.create_watcher_data(transaction.tx_hex(), preimage.tx_hex());
-                    let swpmsg_watcher = SwapWatcherMsg::TakerSwapWatcherMsg(Box::new(watcher_data));
-                    broadcast_swap_message(
-                        &self.ctx,
-                        watcher_topic(&self.r().data.taker_coin),
-                        swpmsg_watcher,
-                        &self.p2p_privkey,
+        // Watchers cannot be used for lightning swaps for now
+        // Todo: Check if watchers can work in some cases with lightning and implement it if it's possible, the watcher will not be able to retrieve the preimage since it's retrieved through the lightning network
+        // Todo: The watcher can retrieve the preimage only if he is running a lightning node and is part of the nodes that routed the taker payment which is a very low probability event that shouldn't be considered
+        let maybe_maker_hex = self.r().maker_payment.as_ref().unwrap().tx_hex.clone();
+        if let Some(maker_hex) = maybe_maker_hex {
+            if let Some(taker_hex) = tx_hex {
+                if self.ctx.use_watchers() {
```

Comment:
looks good, thanks for the suggestion :). But this will be changed due to this https://github.com/KomodoPlatform/atomicDEX-API/pull/1497#discussion_r1003262201.